src/main/haskell/kore/src/Kore/Step/OrOfExpandedPattern.hs

```diff
@@ -337,14 +336,23 @@ crossProductGenericF
 crossProductGenericF joiner (MultiOr first) (MultiOr second) =
     MultiOr <$> sequenceA (joiner <$> first <*> second)
 
-{-| 'merge' merges two 'OrOfExpandedPattern'.
--}
+{- | Merge two disjunctions of patterns.
+
+The arguments are simplified by filtering on @\\top@ and @\\bottom@.  The
+idempotency property of disjunction (@\\or(φ,φ)=φ@) is applied to remove
+duplicated patterns from the result.
+
+See also: 'filterOr'
+
+ -}
 merge
-    :: OrOfExpandedPattern level variable
+    :: (Ord level, Ord (variable level))
+    => OrOfExpandedPattern level variable
     -> OrOfExpandedPattern level variable
     -> OrOfExpandedPattern level variable
-merge patts1 patts2 =
-    filterOr (mergeGeneric patts1 patts2)
+merge (MultiOr patts1) (MultiOr patts2) =
+    (filterOr . MultiOr . Set.toList)
+        (Set.fromList patts1 <> Set.fromList patts2)
```

Comment:
1. Could you add a note that this does not work as expected when, e.g., the patterns contain `\exists` and `\forall` statements.

2. It seems to me that this idempotency thing should also apply to, say, `make` and `traverseWithPairs` (in general: to all or most functions that call `filterOr`).

3. Could you add tests? At least one or more unit-tests for for OrOfEXpandedPattern, but maybe also a test for the original problem (`(a or a)=a` simplification or whatever).